### ISSUE TITLE
Speed up VSCode loading

### ISSUE THREAD

**Issue by dominiksalvet** [yzhang-gh/vscode-markdown#1195]
### Proposal

I propose not loading the extension before detecting an MD language file actually open. Currently, it is activated when there is an MD document in the workspace, which, for open-source projects, is almost always. But I may open VSCode and want to work on code only, and I may want it as soon as possible (so not loading this extension may help). It is fine to load this extension when I open an MD file first time in a session.

Please, would it be possible to change it? If not, what is the reasoning behind this decision?

### Other information

Related screenshot from the extension feature contributions:

![image](https://user-images.githubusercontent.com/50219597/201445857-73d4bed1-532c-468b-b210-27b022aa6360.png)

**Comment by yzhang-gh** [comment]
Thanks for the feedback.

Since the start of this extension in 2016, its activation events have been changed from `*` to only `onLanguage:markdown` and then more conditions as you have seen (and recently we added `onLanguage:rmarkdown`). The reason is simple, VS Code needs a while to load the extension. If it is only activated upon the first opening of a Markdown file, I have seen many reports that this extension was not ready when they pressed, e.g., <kbd>Enter</kbd>/<kbd>Backspace</kbd> keys. I would say it is the most mentioned issue in this repository,

<img src="https://user-images.githubusercontent.com/7588612/201454057-01d8b85a-8ee4-4561-b4cb-103cfc2449b3.png" width=500>

For your concern, this extension does nothing in the background so I don't think it is a problem to allow an extra ~200ms pre-activation even if it is not going to be used. If you do see some (obvious) slow-down because of this extension, feel free to tell me.
